### PR TITLE
refactor(NgClass): cleanup, readability

### PR DESCRIPTION
**Pull request by vicb**
@IgorMinar please review, I consider this PR docs.

Mostly refactoring but also fixes minor issues, wrong types, ...

### CODE REVIEW

File: modules/@angular/common/test/directives/ng_class_spec.ts [hunk 1]
```diff
@@ -6,373 +6,319 @@
...
-import {ListWrapper, StringMapWrapper} from '../../src/facade/collection';
-
-function detectChangesAndCheck(fixture: ComponentFixture<any>, classes: string) {
-  fixture.detectChanges();
-  expect(fixture.debugElement.children[0].nativeElement.className).toEqual(classes);
-}
-
 export function main() {
-  describe('binding to CSS class list', () => {
+  ddescribe('binding to CSS class list', () => {
```
Comment (pkozlowski-opensource) [inline modules/@angular/common/test/directives/ng_class_spec.ts]: `ddescribe` = > we should be failing build on this to automatically catch errors like this.

Comment (vicb) [inline modules/@angular/common/test/directives/ng_class_spec.ts]: Good catch.
Agree, can you create an issue ?  Fell free to work on it also :)

Comment (pkozlowski-opensource) [inline modules/@angular/common/test/directives/ng_class_spec.ts]: There is an issue for this already :-) I will PR a fix later today when I'm at home with access to mac to do the shrinkwrap magic

Comment (vicb) [inline modules/@angular/common/test/directives/ng_class_spec.ts]: Can play the magician of you need

File: modules/@angular/common/test/directives/ng_class_spec.ts [hunk 2]
```diff
@@ -6,373 +6,328 @@
...
-function detectChangesAndCheck(fixture: ComponentFixture<any>, classes: string) {
-  fixture.detectChanges();
-  expect(fixture.debugElement.children[0].nativeElement.className).toEqual(classes);
-}
-
 export function main() {
   describe('binding to CSS class list', () => {
+    let fixture: ComponentFixture<any>;
+
+    function _detectChangesAndExpectClassName(classes: string): void {
```
Comment (IgorMinar) [inline modules/@angular/common/test/directives/ng_class_spec.ts]: the _ prefix is unnecessary and only makes the tests harder to read. can you remove?

File: modules/@angular/common/test/directives/ng_class_spec.ts [hunk 3]
```diff
@@ -6,373 +6,328 @@
...
+      fixture.detectChanges();
+      expect(fixture.debugElement.children[0].nativeElement.className).toEqual(classes);
+    }
+
+    function _cmp(): TestComponent { return fixture.debugElement.componentInstance; }
```
Comment (IgorMinar) [inline modules/@angular/common/test/directives/ng_class_spec.ts]: `getComponentInstance` would make the tests more readable IMO

File: modules/@angular/facade/src/collection.ts [hunk 4]
```diff
@@ -149,18 +149,20 @@ export class StringMapWrapper {
...
+    const k2 = Object.keys(m2);
+
     if (k1.length != k2.length) {
       return false;
     }
-    var key: any /** TODO #???? */;
-    for (var i = 0; i < k1.length; i++) {
-      key = k1[i];
+
+    for (let i = 0; i < k1.length; i++) {
```
Comment (IgorMinar) [inline modules/@angular/facade/src/collection.ts]: use for of: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of

Comment (vicb) [inline modules/@angular/facade/src/collection.ts]: don't want to impact perf on shared code

**Pull request merged**